src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -67,6 +67,11 @@ internal partial class Http2Connection : IHttp2StreamLifetimeHandler, IHttpHeade
         private int _gracefulCloseInitiator;
         private int _isClosed;
 
+        private Http2StreamStack _streamPool;
+
+        internal const int InitialStreamPoolSize = 20;
+        internal const int MaxStreamPoolSize = 50;
```

Comment:
Nit: MaxStreamPoolSize might as well be `InitialStreamPoolSize * 2^n` for some integer N.

Reply:
I'll see what the perf numbers look like with 10 and 40 as the params.

Reply:
InitialStreamPoolSize shouldn't have any effect on the benchmarks because you'd exceed that during the warm-up period. It's also a cheep array expansion if it's too low.